File: hw/total/white_box/BST_insert/code/python/insert.py
from BinNode import BinNode
from Tree import Tree


def remove(T,z):
    # insert BinNode z into Tree T
    y = None
    x = T.root
    while x != None:
        y = x
        if x.v < z.v:
            x = x.lc
        elif x.v == z.v:
            return False
        else:
            x = x.rc
    z.set_p(y)
    if y == None:
        T.root = z
        return True
    else:
        if z.v > y.v:
            y.set_rc(z)
        else:
            y.set_lc(z)
        return True

File: hw/total/white_box/BST_insert/code/java/coverage/insert.java
public class insert {
	public boolean insert(Tree T, BinNode z) {
		BinNode now = T.root;
		if (now == null) {
			T.root = z;
			return true;
		}
		BinNode parent = now;
		while (now != null) {
			parent = now;
			if (now.v > z.v)
				now = now.lc;
			else if (now.v < z.v)
				now = now.rc;
			else
				return false;
		}
		z.set_p(parent);
		if (parent.v > z.v)
			parent.lc = z;
		else
			parent.rc = z;
		return true;
	}
}
